core/src/view/view.cpp

```diff
@@ -226,19 +217,19 @@ void View::updateMatrices() {
     // the "top" face of the view frustum with the ground plane
 
     // NOTE: far here can go to infinity and hence a std::min below!
-    float far = m_pos.z / cos(m_pitch + 0.5 * fovy);
+    float far = 2.f * m_pos.z / cos(m_pitch + 0.5 * fovy);
```

Comment:
Also ... we have not noticed before but far can go negative.
We should use fabs here.